qiita_ware/analysis_pipeline.py

```diff
@@ -89,6 +89,45 @@ def _finish_analysis(analysis, **kwargs):
         analysis.status = "error"
 
 
+def _generate_analysis_tgz(analysis, **kwargs):
+    """Generates the analysis tgz
+
+    Parameters
+    ----------
+    analysis: Analysis
+        Analysis to generate the tgz command from
+    kwargs : ignored
+        Necessary to have in parameters to support execution via moi.
+    """
+    from qiita_db.sql_connection import TRN
+    from qiita_db.util import get_mountpoint, get_mountpoint_path_by_id
+    from os.path import join
+    from subprocess import Popen, PIPE
+
+    fps_ids = analysis.all_associated_filepath_ids
+    with TRN:
```

Comment:
This is `qiita_ware`, in `qiita_ware` shouldn't be any SQL so please remove this from here. My recommendation will be to add a new method to analysis `generate_tgz` that performs this task. This way, everything regarding the data stored in analysis is managed internally by the analysis object.